driver-core/src/main/com/mongodb/client/model/expressions/DateExpression.java

```diff
@@ -20,5 +20,18 @@
  * Expresses a date value.
  */
 public interface DateExpression extends Expression {
+    IntegerExpression year();
+    IntegerExpression month();
+    IntegerExpression dayOfMonth();
+    IntegerExpression dayOfWeek();
+    IntegerExpression dayOfYear();
+    IntegerExpression hour();
+    IntegerExpression minute();
+    IntegerExpression second();
+    IntegerExpression week();
+    IntegerExpression millisecond();
+
+    StringExpression dateToString();
+    StringExpression dateToString(StringExpression format, StringExpression timezone);
```

Comment:
I don't recall if we've had a discussion about how to handle expressions with multiple optional parameters.  In this case both format and timezone are independently optional and it seems necessary to support that in the API.

Reply:
Ping on this.

Reply:
We appear to be spared in this particular case, because omitting one of the options would be confused or at least confusing, see: https://github.com/mongodb/mongo-java-driver/pull/1045#discussion_r1028655799 . That is, they should be mandatory, even if the server docs disagree.

My original thought (before @stIncMale pointed out the above) was to err on the side of being more explicit, especially in cases where the method is infrequent, and where implicit values might sometimes lead to confusion. Of course, this might not always work or make sense, but might help in other cases as well. It is also possible to sometimes get rid of optionals by moving them out (as was the case with limit on filter). So the general approach is to try to get rid of them in some way or another. This seems to lead to a cleaner API.